parsley-debug/shared/src/main/scala/parsley/debugger/util/Collector.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Parsley Contributors <https://github.com/j-mie6/Parsley/graphs/contributors>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+package parsley.debugger.util
+
+import parsley.Parsley
+import parsley.debugger.internal.Rename
+import parsley.token.Lexer
+
+import parsley.internal.deepembedding.frontend.LazyParsley
+
+/** Attempt to collect all the fields in a class or object that contain a
+  * parser of type [[parsley.Parsley]], or from a [[parsley.token.Lexer]].
+  *
+  * This information is used later in the debug tree-building process to rename certain parsers
+  * so that they do not end up being named things like "packageanon".
+  *
+  * You only need to run this once per parser-holding object.
```

Comment:
`@since 4.5.0`

Reply:
Added in [0b3309d](https://github.com/j-mie6/parsley/pull/219/commits/0b3309d533215e4f3094a4fc4019491971b00d9e)